lib/spack/spack/fetch_strategy.py

```diff
@@ -776,6 +776,10 @@ def git(self):
         if not self._git:
             self._git = which('git', required=True)
 
+            # Disable advice for a quieter fetch
+            self._git.add_default_arg('-c')
+            self._git.add_default_arg('advice.detachedHead=false')
```

Comment:
Can we check the minimum git version required for this option? I was about to merge the PR, but checked before doing that our `centos6` image. There we have:
```console
# git --version
git version 1.7.1
```
and the option is not present.

Reply:
Any idea where to find this information? I have git 1.7.12.4 installed on a system and `man git-config` has `detachedHead` in it. So it was added in some version in that range: 1.7.1-1.7.12. Does your git 1.7.1 have a `-c` config flag but not `detachedHead` advice?

Reply:
Here's what I did:
```console
$ docker run -it --rm spack/centos6
[root@e871d7607258 ~]# git --version
git version 1.7.1
[root@e871d7607258 ~]# git -c advice.detachedHead=false log
Unknown option: -c
[ ... ]
```